src/main/java/org/seqdoop/hadoop_bam/util/BGZFBlockIndexer.java

```
@@ -95,36 +89,36 @@ public BGZFBlockIndexer(int g) {
 	}
 
 	private void index(final File file) throws IOException {
-		final InputStream in = new FileInputStream(file);
+		try (final InputStream in = new FileInputStream(file)) {
 
-		final OutputStream out = new BufferedOutputStream(
-			new FileOutputStream(file.getPath() + ".bgzfi"));
+			try (final OutputStream out = new BufferedOutputStream(
+					new FileOutputStream(file.getPath() + ".bgzfi"))) {
 
-		final LongBuffer lb =
-			byteBuffer.order(ByteOrder.BIG_ENDIAN).asLongBuffer();
+				final LongBuffer lb =
+						byteBuffer.order(ByteOrder.BIG_ENDIAN).asLongBuffer();
 
-		long prevPrint = 0;
-		pos = 0;
+				long prevPrint = 0;
+				pos = 0;
 
-		for (int i = 0;;) {
-			if (!skipBlock(in))
-				break;
+				for (int i = 0; ; ) {
+					if (!skipBlock(in))
+						break;
 
-			if (++i == granularity) {
-				i = 0;
-				lb.put(0, pos);
-				out.write(byteBuffer.array(), 2, 6);
+					if (++i == granularity) {
+						i = 0;
+						lb.put(0, pos);
+						out.write(byteBuffer.array(), 2, 6);
 
-				if (pos - prevPrint >= PRINT_EVERY) {
-					System.out.print("-");
-					prevPrint = pos;
+						if (pos - prevPrint >= PRINT_EVERY) {
+							System.out.print("-");
+							prevPrint = pos;
+						}
+					}
 				}
-			}
-		}
-		lb.put(0, file.length());
-		out.write(byteBuffer.array(), 2, 6);
-		out.close();
-		in.close();
+				lb.put(0, file.length());
+				out.write(byteBuffer.array(), 2, 6);
+			} catch (IOException e) {}
```

I wouldn't silently catch these exceptions.